.github/workflows/pr-submodule-branch.yml

```diff
@@ -0,0 +1,35 @@
+name: Submodule Branch Check
+on:
+  pull_request:
+    types: ['opened', 'edited', 'reopened', 'synchronize']
+  push:
+    branches:
+      - develop
+      - 'release/**'
+      - staging
+
+jobs:
+  submodule-branch:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          submodules: recursive
+
+      - uses: cachix/install-nix-action@v22
+      - name: Pre-populate nix-shell
+        run: |
+            export NIX_PATH=nixpkgs=$(jq '.nixpkgs.url' nix/sources.json -r)
```

Comment:
tabbing here is different than below, let's be consistent

Reply:
fixed